// angular-email-editor/esm2015/lib/email-editor.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { loadScript } from './loadScript';
import pkg from './source.json';
let lastEditorId = 0;
export class EmailEditorComponent {
    constructor() {
        this.options = {};
        this.minHeight = '500px';
        this.loaded = new EventEmitter();
        this.editorId = `editor-${++lastEditorId}`;
    }
    ngOnInit() { }
    ngAfterViewInit() {
        loadScript(this.loadEditor.bind(this));
    }
    loadEditor() {
        const options = this.options || {};
        if (this.projectId) {
            options.projectId = this.projectId;
        }
        if (this.tools) {
            options.tools = this.tools;
        }
        if (this.appearance) {
            options.appearance = this.appearance;
        }
        if (this.locale) {
            options.locale = this.locale;
        }
        this.editor = unlayer.createEditor(Object.assign({}, options, { id: this.editorId, displayMode: 'email', source: {
                name: pkg.name,
                version: pkg.version,
            } }));
        this.loaded.emit({});
    }
    loadDesign(data) {
        this.editor.loadDesign(data);
    }
    saveDesign(cb) {
        this.editor.saveDesign(cb);
    }
    exportHtml(cb) {
        this.editor.exportHtml(cb);
    }
}
EmailEditorComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'email-editor',
                template: "<div [id]=\"editorId\" class=\"unlayer-editor\" [style.min-height]=\"minHeight\"></div>",
                styles: [":host{flex:1;display:flex}.unlayer-editor{flex:1;display:flex}"]
            }] }
];
/** @nocollapse */
EmailEditorComponent.ctorParameters = () => [];
EmailEditorComponent.propDecorators = {
    editorId: [{ type: Input }],
    options: [{ type: Input }],
    projectId: [{ type: Input }],
    tools: [{ type: Input }],
    appearance: [{ type: Input }],
    locale: [{ type: Input }],
    minHeight: [{ type: Input }],
    loaded: [{ type: Output }]
};
