// angular-email-editor/fesm5/angular-email-editor.js
import { Injectable, NgModule, defineInjectable, EventEmitter, Component, Input, Output } from '@angular/core';
import { __assign } from 'tslib';

var EmailEditorService = /** @class */ (function () {
    function EmailEditorService() {
    }
    EmailEditorService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    EmailEditorService.ctorParameters = function () { return []; };
    EmailEditorService.ngInjectableDef = defineInjectable({ factory: function EmailEditorService_Factory() { return new EmailEditorService(); }, token: EmailEditorService, providedIn: "root" });
    return EmailEditorService;
}());

var scriptUrl = '//editor.unlayer.com/embed.js?2';
var callbacks = [];
var loaded = false;
var isScriptInjected = function () {
    var scripts = document.querySelectorAll('script');
    var injected = false;
    scripts.forEach(function (script) {
        if (script.src.includes(scriptUrl)) {
            injected = true;
        }
    });
    return injected;
};
var addCallback = function (callback) {
    callbacks.push(callback);
};
var runCallbacks = function () {
    if (loaded) {
        var callback = void 0;
        while ((callback = callbacks.shift())) {
            callback();
        }
    }
};
var loadScript = function (callback) {
    addCallback(callback);
    if (!isScriptInjected()) {
        var embedScript = document.createElement('script');
        embedScript.setAttribute('src', scriptUrl);
        embedScript.onload = function () {
            loaded = true;
            runCallbacks();
        };
        document.head.appendChild(embedScript);
    }
    else {
        runCallbacks();
    }
};

var name = "angular-email-editor";
var version = "0.7.1";
var peerDependencies = {
	"@angular/common": "^7.0.0",
	"@angular/core": "^7.0.0"
};
var pkg = {
	name: name,
	version: version,
	peerDependencies: peerDependencies
};

var lastEditorId = 0;
var EmailEditorComponent = /** @class */ (function () {
    function EmailEditorComponent() {
        this.options = {};
        this.minHeight = '500px';
        this.loaded = new EventEmitter();
        this.editorId = "editor-" + ++lastEditorId;
    }
    EmailEditorComponent.prototype.ngOnInit = function () { };
    EmailEditorComponent.prototype.ngAfterViewInit = function () {
        loadScript(this.loadEditor.bind(this));
    };
    EmailEditorComponent.prototype.loadEditor = function () {
        var options = this.options || {};
        if (this.projectId) {
            options.projectId = this.projectId;
        }
        if (this.tools) {
            options.tools = this.tools;
        }
        if (this.appearance) {
            options.appearance = this.appearance;
        }
        if (this.locale) {
            options.locale = this.locale;
        }
        this.editor = unlayer.createEditor(__assign({}, options, { id: this.editorId, displayMode: 'email', source: {
                name: pkg.name,
                version: pkg.version,
            } }));
        this.loaded.emit({});
    };
    EmailEditorComponent.prototype.loadDesign = function (data) {
        this.editor.loadDesign(data);
    };
    EmailEditorComponent.prototype.saveDesign = function (cb) {
        this.editor.saveDesign(cb);
    };
    EmailEditorComponent.prototype.exportHtml = function (cb) {
        this.editor.exportHtml(cb);
    };
    EmailEditorComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'email-editor',
                    template: "<div [id]=\"editorId\" class=\"unlayer-editor\" [style.min-height]=\"minHeight\"></div>",
                    styles: [":host{flex:1;display:flex}.unlayer-editor{flex:1;display:flex}"]
                }] }
    ];
    /** @nocollapse */
    EmailEditorComponent.ctorParameters = function () { return []; };
    EmailEditorComponent.propDecorators = {
        editorId: [{ type: Input }],
        options: [{ type: Input }],
        projectId: [{ type: Input }],
        tools: [{ type: Input }],
        appearance: [{ type: Input }],
        locale: [{ type: Input }],
        minHeight: [{ type: Input }],
        loaded: [{ type: Output }]
    };
    return EmailEditorComponent;
}());

var EmailEditorModule = /** @class */ (function () {
    function EmailEditorModule() {
    }
    EmailEditorModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [EmailEditorComponent],
                    imports: [],
                    exports: [EmailEditorComponent]
                },] }
    ];
    return EmailEditorModule;
}());

/*
 * Public API Surface of email-editor
 */

/**
 * Generated bundle index. Do not edit.
 */

export { EmailEditorService, EmailEditorComponent, EmailEditorModule };
