// angular-email-editor/esm5/lib/email-editor.component.js
import * as tslib_1 from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { loadScript } from './loadScript';
import pkg from './source.json';
var lastEditorId = 0;
var EmailEditorComponent = /** @class */ (function () {
    function EmailEditorComponent() {
        this.options = {};
        this.minHeight = '500px';
        this.loaded = new EventEmitter();
        this.editorId = "editor-" + ++lastEditorId;
    }
    EmailEditorComponent.prototype.ngOnInit = function () { };
    EmailEditorComponent.prototype.ngAfterViewInit = function () {
        loadScript(this.loadEditor.bind(this));
    };
    EmailEditorComponent.prototype.loadEditor = function () {
        var options = this.options || {};
        if (this.projectId) {
            options.projectId = this.projectId;
        }
        if (this.tools) {
            options.tools = this.tools;
        }
        if (this.appearance) {
            options.appearance = this.appearance;
        }
        if (this.locale) {
            options.locale = this.locale;
        }
        this.editor = unlayer.createEditor(tslib_1.__assign({}, options, { id: this.editorId, displayMode: 'email', source: {
                name: pkg.name,
                version: pkg.version,
            } }));
        this.loaded.emit({});
    };
    EmailEditorComponent.prototype.loadDesign = function (data) {
        this.editor.loadDesign(data);
    };
    EmailEditorComponent.prototype.saveDesign = function (cb) {
        this.editor.saveDesign(cb);
    };
    EmailEditorComponent.prototype.exportHtml = function (cb) {
        this.editor.exportHtml(cb);
    };
    EmailEditorComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'email-editor',
                    template: "<div [id]=\"editorId\" class=\"unlayer-editor\" [style.min-height]=\"minHeight\"></div>",
                    styles: [":host{flex:1;display:flex}.unlayer-editor{flex:1;display:flex}"]
                }] }
    ];
    /** @nocollapse */
    EmailEditorComponent.ctorParameters = function () { return []; };
    EmailEditorComponent.propDecorators = {
        editorId: [{ type: Input }],
        options: [{ type: Input }],
        projectId: [{ type: Input }],
        tools: [{ type: Input }],
        appearance: [{ type: Input }],
        locale: [{ type: Input }],
        minHeight: [{ type: Input }],
        loaded: [{ type: Output }]
    };
    return EmailEditorComponent;
}());
export { EmailEditorComponent };
